### PR TITLE
First Refactoring of obj Plug-In

### PR DESCRIPTION
**Pull request by zliang-akamai**
Type annotations, simplified logics, simplified functions and function calls

### CODE REVIEW

File: linodecli/plugins/obj.py [hunk 1]
```diff
@@ -798,50 +819,46 @@ def call(args, context):  # pylint: disable=too-many-branches,too-many-statement
...
-    parsed, args = parser.parse_known_args(args)
 
-    # don't mind --no-defaults if it's there; the top-level parser already took care of it
-    try:
-        args.remove("--no-defaults")
-    except ValueError:
-        pass
+def call(
+    args: List[str], context: PluginContext
+):  # pylint: disable=too-many-branches,too-many-statements
```
Comment (jriddle-linode) [inline linodecli/plugins/obj.py:827]: i would like to remove these pylint disables when possible, i threw them up when adding lint so we could ignore the errors while we refactored, but now hopefully our refactoring fixes these errors and they don't need to be disabled

Comment (zliang-akamai) [reply linodecli/plugins/obj.py:827]: Thank you for the suggestion! Let me try if the linter is still reporting errors. If the errors are still there, may I keep this disables and fix them in the next PR? I would like to keep each PR small and easier for reviews.

Comment (zliang-akamai) [reply linodecli/plugins/obj.py:827]: Unfortunately, it is still reporting the errors...

File: linodecli/plugins/obj.py [hunk 2]
```diff
@@ -798,50 +819,46 @@ def call(args, context):  # pylint: disable=too-many-branches,too-many-statement
...
+    """
+    This is called when the plugin is invoked
+    """
+    if not HAS_BOTO:
+        # we can't do anything - ask for an install
+        pip_version = "pip3" if sys.version[0] == 3 else "pip"
```
Comment (lgarber-akamai) [inline linodecli/plugins/obj.py:833]: Wondering if we can drop this expression now that Python 2 support is dropped.

File: linodecli/plugins/obj.py [hunk 3]
```diff
@@ -798,50 +819,46 @@ def call(args, context):  # pylint: disable=too-many-branches,too-many-statement
...
     access_key, secret_key = (
-        os.environ.get(ENV_ACCESS_KEY_NAME, None),
-        os.environ.get(ENV_SECRET_KEY_NAME, None),
+        os.getenv(ENV_ACCESS_KEY_NAME, None),
+        os.getenv(ENV_SECRET_KEY_NAME, None),
     )
 
     if not "--help" in args:
-        if access_key and not secret_key or secret_key and not access_key:
+        if bool(access_key) != bool(secret_key):
```
Comment (lgarber-akamai) [inline linodecli/plugins/obj.py:861]: I love how this logic cleans up :tada:

**Review by lgarber-akamai — APPROVED** [review]
Clean!

File: linodecli/plugins/obj.py (hunk 2 above)
Comment (zliang-akamai) [reply linodecli/plugins/obj.py:833]: Yea, that's on my TODO list.

File: linodecli/plugins/obj.py (hunk 3 above)
Comment (zliang-akamai) [reply linodecli/plugins/obj.py:861]: Thank you! ^_^

**Review by jriddle-linode — APPROVED** [review]

**Pull request merged**